requests/models.py

```diff
@@ -414,7 +414,7 @@ def prepare_body(self, data, files, json=None):
         content_type = None
         length = None
 
-        if json is not None:
+        if data == {} and json is not None:
```

Comment:
This line introduce an other issue, when the request is created with the models.Request object.
In this case, if we instanciate the Request object only with `json` argument (no `data`), then `data` is initialized with an empty list (not dict).

Then when the preparation process arrive to the previous comparison, json is never taken into account (`[] != {}`)